### PR TITLE
fixing issue 110 by using functor instead of inner function

### PR DESCRIPTION
**Pull request by martinResearch**
Fixing issue when pickling described  [here](https://github.com/r1chardj0n3s/parse/issues/110)
Pickle does not work with inner function . The solution proposed here is to use a functor.
It would be good to add some unit tests to test that the parsers can be pickled.

### CODE REVIEW

File: parse.py [hunk 1]
```diff
@@ -513,10 +513,13 @@ def int_convert(base=None):
     it overrides the default base of 10.
 
     It may also have other non-numeric characters that we can ignore.
-    '''
-    CHARS = '0123456789abcdefghijklmnopqrstuvwxyz'
+    '''    
+
+    def __init__(self, base=None):
+        self.base = base
+        self.CHARS = '0123456789abcdefghijklmnopqrstuvwxyz'
```
Comment (r1chardj0n3s) [inline parse.py:520]: CHARS doesn't need to move to be an instance attribute - leaving it where it was will make it a class attribute.

**Comment by martinResearch** [comment]
I made the change for CHARS. I also added another functor to avoid the use of lambda functions that cannot be pickled either

**Comment by martinResearch** [comment]
Hi, is there anything left to do to in order to complete this pull request ?

**Comment by r1chardj0n3s** [comment]
Thanks for the change, I've just written a test though to make sure the fix is good.

**Pull request merged**